### PR TITLE
Added explicit load and store actions to all SetRenderTarget commands.

### PR DESCRIPTION
**Pull request by aleks01010101**

### CODE REVIEW

File: ScriptableRenderPipeline/LightweightPipeline/LWRP/LightweightPipeline.cs [hunk 1]
```diff
@@ -1063,10 +1065,14 @@ private void EndForwardRendering(ref ScriptableRenderContext context, FrameRende
...
                 // If PostProcessing is enabled, it is already blit to CameraTarget.
                 if (!CoreUtils.HasFlag(renderingConfig, FrameRenderingConfiguration.PostProcess))
+                {
                     Blit(cmd, renderingConfig, m_CurrCameraColorRT, BuiltinRenderTextureType.CameraTarget, blitMaterial);
+                    skipSetRT = true;
+                }
             }
 
-            SetRenderTarget(cmd, BuiltinRenderTextureType.CameraTarget);
+            if (!skipSetRT)
```
Comment (phi-lira) [inline ScriptableRenderPipeline/LightweightPipeline/LWRP/LightweightPipeline.cs]: Seems like we can avoid this skipSetRT if we add the SetRenderTarget as else to the else/if clause above and rework the else/if a bit. 

else if (intermediateTexture && !PostProcess)
    blitMaterial = (stereo) ? ...

File: ScriptableRenderPipeline/LightweightPipeline/LWRP/LightweightPipeline.cs [hunk 2]
```diff
@@ -1130,22 +1136,32 @@ private void RenderPostProcess(CommandBuffer cmd, RenderTargetIdentifier source,
                 m_CameraPostProcessLayer.Render(m_PostProcessRenderContext);
         }
 
-        private void SetRenderTarget(CommandBuffer cmd, RenderTargetIdentifier colorRT, ClearFlag clearFlag = ClearFlag.None)
+        private void SetRenderTarget(CommandBuffer cmd, RenderTargetIdentifier colorRT, RenderBufferLoadAction loadAction, RenderBufferStoreAction storeAction, ClearFlag clearFlag = ClearFlag.None)
         {
-            int depthSlice = (m_IntermediateTextureArray) ? -1 : 0;
-            CoreUtils.SetRenderTarget(cmd, colorRT, clearFlag, CoreUtils.ConvertSRGBToActiveColorSpace(m_CurrCamera.backgroundColor), 0, CubemapFace.Unknown, depthSlice);
+            if (m_IntermediateTextureArray)
+                CoreUtils.SetRenderTarget(cmd, colorRT, clearFlag, CoreUtils.ConvertSRGBToActiveColorSpace(m_CurrCamera.backgroundColor), 0, CubemapFace.Unknown, -1);
```
Comment (phi-lira) [inline ScriptableRenderPipeline/LightweightPipeline/LWRP/LightweightPipeline.cs]: Doesn't it make sense to have the LOAD/STORE actions set here?

File: ScriptableRenderPipeline/LightweightPipeline/LWRP/LightweightPipeline.cs [hunk 3]
```diff
@@ -1130,22 +1136,32 @@ private void RenderPostProcess(CommandBuffer cmd, RenderTargetIdentifier source,
...
             {
-                SetRenderTarget(cmd, colorRT, clearFlag);
+                SetRenderTarget(cmd, colorRT, colorLoadAction, colorStoreAction, clearFlag);
                 return;
             }
 
-            int depthSlice = (m_IntermediateTextureArray) ? -1 : 0;
-            CoreUtils.SetRenderTarget(cmd, colorRT, depthRT, clearFlag, CoreUtils.ConvertSRGBToActiveColorSpace(m_CurrCamera.backgroundColor), 0, CubemapFace.Unknown, depthSlice);
+            if (m_IntermediateTextureArray)
+                CoreUtils.SetRenderTarget(cmd, colorRT, depthRT, clearFlag, CoreUtils.ConvertSRGBToActiveColorSpace(m_CurrCamera.backgroundColor), 0, CubemapFace.Unknown, -1);
```
Comment (phi-lira) [inline ScriptableRenderPipeline/LightweightPipeline/LWRP/LightweightPipeline.cs]: Same as above

File: ScriptableRenderPipeline/Core/CoreRP/Utilities/CoreUtils.cs [hunk 4]
```diff
@@ -186,6 +186,26 @@ public static void SetRenderTarget(CommandBuffer cmd, RenderTargetIdentifier[] c
             ClearRenderTarget(cmd, clearFlag, clearColor);
         }
 
+        // Explicit load and store actions
+        public static void SetRenderTarget(CommandBuffer cmd, RenderTargetIdentifier buffer, RenderBufferLoadAction loadAction, RenderBufferStoreAction storeAction, ClearFlag clearFlag, Color clearColor)
```
[the remark this answers is not in the document]
Comment (aleks01010101) [reply ScriptableRenderPipeline/Core/CoreRP/Utilities/CoreUtils.cs]: This first has to be done on the engine side. There's no overload in the CommandBuffer API that takes those parameters.

File: ScriptableRenderPipeline/LightweightPipeline/LWRP/LightweightPipeline.cs [hunk 5]
```diff
@@ -390,7 +390,7 @@ private void OpaqueTexturePass(ref ScriptableRenderContext context, FrameRenderi
                     cmd.Blit(m_CurrCameraColorRT, CameraRenderTargetID.opaque);
                     break;
             }
-            SetRenderTarget(cmd, m_CurrCameraColorRT, m_DepthRT);
+            SetRenderTarget(cmd, m_CurrCameraColorRT, depthRT, RenderBufferLoadAction.Load, RenderBufferStoreAction.Store);
```
[the remark this answers is not in the document]
Comment (aleks01010101) [reply ScriptableRenderPipeline/LightweightPipeline/LWRP/LightweightPipeline.cs]: There are two overloads: one that takes l/s actions for color and separate ones for depth and one with color and depth having the same l/s actions.

File: ScriptableRenderPipeline/LightweightPipeline/LWRP/LightweightPipeline.cs (hunk 2 above)
Comment (aleks01010101) [reply ScriptableRenderPipeline/LightweightPipeline/LWRP/LightweightPipeline.cs]: There's no overload on the engine side for this at the moment.

**Pull request merged**